File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeNthFromEnd(head *ListNode, n int) *ListNode {
    counter := 1
    dummy := &ListNode{Val: 0 , Next: head }
    slow := dummy
    fast := head 
    
    for counter <= n {
        fast = fast.Next
        counter++
    }
    
    
    for fast != nil {
        fast = fast.Next
        slow = slow.Next
    }
    slow.Next = slow.Next.Next
    return dummy.Next
    
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let counter = 1
    let dummy = new ListNode(0 , head)
    let slow = dummy 
    let fast = head 

    while(counter <= n ) {
        fast = fast.next 
        counter++
    }

    while(fast !== null ) {
        fast = fast.next
        slow = slow.next
    }

    slow.next = slow.next.next 
    return dummy.next
};